onsstove/technology.py

```diff
@@ -94,6 +94,25 @@ def paf(rr, sfu):
         paf = (sfu * (rr - 1)) / (sfu * (rr - 1) + 1)
 
         return paf
+
+    @staticmethod
+    def discount_factor(self, specs_file):
+        '''
+
+        :param self:
+        :param specs_file: social specs file
+        :return: discount factor to be used for all costs in the net benefit fucntion and the years of analysis
+        '''
+        if specs_file["Start_year"] == specs_file["End_year"]:
+            proj_life = 1
+        else:
+            proj_life = specs_file["End_year"] - specs_file["Start_year"]
+
+        year = np.arange(proj_life)
+
+        discount_factor = (1 + discount_rate_tech) ** year
```

Comment:
were is `discount_rate_tech` defined? if it is an attribute from the class then it should use `self`

Reply:
It is defined in the specs file

Reply:
a ok, then you can make it as ` specs_file["discount_rate_tech"]` or something similar